goodconf/__init__.py

```diff
@@ -200,16 +181,14 @@ def settings_customise_sources(
 
     model_config = GoodConfConfigDict()
 
+    def _load(
+        self, filename: str | None = None, kwargs={}, init_filename: str | None = None
+    ) -> None:
+        CONFIG_CONTEXT.set(filename or init_filename)
+        super().__init__(**kwargs)
+
     def load(self, filename: str | None = None) -> None:
-        """Find config file and set values"""
-        if filename:
-            values = _load_config(filename)
```

Comment:
Even if we are not using my approach with context vars and keep the current approach it might make sense to simply set `._config_file` here instead and leave the actual loading to our config source so everything is in one place.